lib/matplotlib/colors.py

```diff
@@ -1112,7 +1112,8 @@ def _transform(self, a):
         """
         Inplace transformation.
         """
-        masked = np.abs(a) > self.linthresh
+        with np.errstate(invalid="ignore"):
```

Comment:
Why does this throw an invalid warning in the first place? It's not obvious to me that `_transform` should be getting invalid values.

Reply:
You can definitely pass nans in (they're ignored at a later stage, by the low-level path handling machinery).